<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="camera_frame_id" default="camera_link"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="base_frame_id" default="base_link"/>

  <arg name="robot_urdf_file_path" default="$(find adream_simulation)/models/cad_models/urdf/r2d2.urdf"/>
  <arg name="env_urdf_file_path" default="$(find adream_simulation)/models/cad_models/urdf/adream_empty.urdf"/>
  <arg name="cad_models_additional_search_path" default="$(find adream_simulation)/models/cad_models"/>

  <arg name="static_entities_config_filename" default="$(find adream_simulation)/config/environment/adream_static_entities.yaml"/>

  <!-- Camera parameters -->
  <arg name="rgb_image_topic" default="xtion/rgb/image_rect_color"/>
  <arg name="rgb_camera_info_topic" default="xtion/rgb/camera_info"/>
  <arg name="depth_image_topic" default="xtion/depth/image_rect"/>
  <arg name="depth_camera_info_topic" default="xtion/depth/camera_info"/>

  <arg name="use_gui" default="true"/>

  <node name="uwds3_simulation" pkg="uwds3_simulation" type="uwds3_simulation_node.py" args="" output="screen" respawn="false">
    <rosparam subst_value="true">

      rgb_image_topic: $(arg rgb_image_topic)
      rgb_camera_info_topic: $(arg rgb_camera_info_topic)
      depth_image_topic: $(arg depth_image_topic)
      depth_camera_info_topic: $(arg depth_camera_info_topic)

      global_frame_id: $(arg global_frame_id)
      base_frame_id: $(arg base_frame_id)
      camera_frame_id: $(arg camera_frame_id)

      robot_urdf_file_path: $(arg robot_urdf_file_path)
      env_urdf_file_path: $(arg env_urdf_file_path)
      cad_models_additional_search_path: $(arg cad_models_additional_search_path)

      static_entities_config_filename: $(arg static_entities_config_filename)

      use_gui: $(arg use_gui)

    </rosparam>
  </node>

</launch>
